packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts

```diff
@@ -203,6 +210,246 @@ describe('BelongsTo relation', () => {
+      where: {description: 'group 1'},
+    });
+    expect(group1).to.deepEqual([order1]);
+  });
+
+  it('deletes an instance then delete the through model', async () => {
+    await hasManyThroughRepo.create(
+      {
+        description: 'customer 1',
+      },
+      {
+        throughData: {id: 98},
+      },
+    );
+    const anotherHasManyThroughRepo = hasManyThroughFactory(
+      existingCustomerId + 1,
+    );
+    const order2 = await anotherHasManyThroughRepo.create(
+      {
+        description: 'customer 2',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    let orders = await orderRepo.find();
+    let links = await customerOrderLinkRepo.find();
+
+    expect(orders).have.length(2);
+    expect(links).have.length(2);
+
+    await hasManyThroughRepo.delete();
+    orders = await orderRepo.find();
+    links = await customerOrderLinkRepo.find();
+
+    expect(orders).have.length(1);
+    expect(links).have.length(1);
+    expect(orders).to.deepEqual([order2]);
+    expect(links[0]).has.property('orderId', order2.id);
+    expect(links[0]).has.property('customerId', existingCustomerId + 1);
+  });
+
+  it('through models get deleted once the related target gets deleted', async () => {
+    const order1 = await hasManyThroughRepo.create(
+      {
+        description: 'customer 1',
+      },
+      {
+        throughData: {id: 98},
+      },
+    );
+    const anotherHasManyThroughRepo = hasManyThroughFactory(
+      existingCustomerId + 1,
+    );
+    const order2 = await anotherHasManyThroughRepo.create(
+      {
+        description: 'customer 2',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    // when order1 gets deleted, this through instance should be deleted too.
+    const through = await customerOrderLinkRepo.create({
+      id: 1,
+      customerId: existingCustomerId + 1,
+      orderId: order1.id,
+    });
+    let orders = await orderRepo.find();
+    let links = await customerOrderLinkRepo.find();
+
+    expect(orders).have.length(2);
+    expect(links).have.length(3);
+
+    await hasManyThroughRepo.delete();
+
+    orders = await orderRepo.find();
+    orders = await orderRepo.find();
+    links = await customerOrderLinkRepo.find();
+
+    expect(orders).have.length(1);
+    expect(links).have.length(1);
+    expect(orders).to.deepEqual([order2]);
+    expect(links).to.not.containEql(through);
+    expect(links[0]).has.property('orderId', order2.id);
+    expect(links[0]).has.property('customerId', existingCustomerId + 1);
+  });
+
+  it('patches instances that belong to the same source model (same source fk)', async () => {
+    const order1 = await hasManyThroughRepo.create(
+      {
+        description: 'group 1',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    const order2 = await hasManyThroughRepo.create(
+      {
+        description: 'group 1',
+      },
+      {
+        throughData: {id: 98},
```

Comment:
I like the fact that it's possible to specify "throughData" when creating new instances of the related model :+1: